idealingua-v1/idealingua-v1-runtime-rpc-http4s/src/main/scala/izumi/idealingua/runtime/rpc/http4s/WithWebsocketClientContext.scala

```diff
@@ -246,20 +248,22 @@ class WebsocketClientContextImpl[C <: Http4sContext]
   protected[http4s] val pingStream: fs2.Stream[MonoIO, WebSocketFrame] =
     fs2.Stream.awakeEvery[MonoIO](pingTimeout) >> fs2.Stream(Ping())
 
-  override def finish(): Unit = {
-    onWsSessionClosed()
-    Quirks.discard(wsSessionStorage.deleteClient(sessionId))
-    listeners.foreach { listener =>
-      listener.onSessionClosed(id)
-    }
-    requestState.clear()
+  override def finish(): C#MonoIO[Unit] = {
+    onWsSessionClosed() *>
+    wsSessionStorage.deleteClient(sessionId) *>
+    F.traverse_(listeners) {
+      listener =>
+        listener.onSessionClosed(id)
+    } *>
```

Comment:
@Caparow Would be nicer for readability if you use for-comprehensions instead of `*>` everywhere when there's more than one `*>` operator...